### PR TITLE
fix(compiler): struct literals allow nonexistent fields

### LINKED ISSUE #1881: struct literal accepts non existent fields
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

I tried to compile this
```wing
// wing
bring cloud;

let x = cloud.ApiResponse {
    status: 200,
    body: "Hello, World!",
    badger: "badger"
};
```

### I expected this:

compilation error:
"Field badger not part of struct cloud.ApiResponse"

### Instead, this happened:

It compiled.
This bug was introduced here: #1864
I also don't understand why in #1864 we duplicate the field name in the Struct ast struct: https://github.com/winglang/wing/blob/a56e603caf2c739eab4ed20feb3a2e43cb82568e/libs/wingc/src/ast.rs#L363
@Chriscbr ?

### Component:

Compiler

### Wing Version:

_No response_

### Node.js Version:

_No response_

### Platform(s):

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #1881

Misc:
- Changed AST struct fields from BTreeMap to IndexMap after feedback from @yoav-steinberg (and updated other usages of BTreeMap as well for consistency)

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingc/src/ast.rs [hunk 1]
```diff
@@ -360,7 +360,7 @@ pub enum ExprKind {
 	StructLiteral {
 		type_: TypeAnnotation,
 		// We're using an ordered map implementation to guarantee deterministic compiler output. See discussion: https://github.com/winglang/wing/discussions/887.
```
Comment (Chriscbr) [inline libs/wingc/src/ast.rs:362]: Fix this comment

File: libs/wingc/src/type_check.rs [hunk 2]
```diff
@@ -1451,16 +1460,16 @@ impl<'a> TypeChecker<'a> {
...
 				for (name, kind, _info) in st.env.iter(true) {
 					let field_type = kind
 						.as_variable()
 						.expect("Expected struct field to be a variable in the struct env")
 						.type_;
-					match fields.get(&name) {
-						Some((_, field_exp)) => {
-							let t = self.type_check_exp(field_exp, env, statement_idx, context);
-							self.validate_type(t, field_type, field_exp);
+					match fields.get(&Symbol::global(&name)) {
```
Comment (MarkMcCulloh) [inline libs/wingc/src/type_check.rs:1469]: Do we want to go as far as implementing [Equivalent](https://docs.rs/indexmap/latest/indexmap/trait.Equivalent.html) for Symbol so we can do this with a String?

File: tools/hangar/project.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
         "examples-tests"
       ],
       "outputs": [
-        "{projectRoot}/src/test_corpus/**/*}"
+        "{projectRoot}/src/test_corpus/**/*"
```
Comment (MarkMcCulloh) [inline tools/hangar/project.json:15]: 🙇

**Review by MarkMcCulloh — APPROVED** [review]

File: libs/wingc/src/type_check.rs (hunk 2 above)
Comment (Chriscbr) [reply libs/wingc/src/type_check.rs:1469]: oh, interesting. Probably a good idea

**Review by yoav-steinberg — APPROVED** [review]

**Pull request merged**